### PR TITLE
chore: dropped pub fields from config module

### LINKED ISSUE #2256: refactor: Drop `pub` from all the fields of ConfigModule
**Technical Requirements**
- Derive getters to get access by reference to the internal fields.
- There is only one setter usage for ConfigModule that can be changed to use `config_module::Extension` which has pub fields.

### PR DESCRIPTION
**Pull request by dekkku**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #2256 
/claim #2256

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/config/config_module.rs [hunk 1]
```diff
@@ -15,13 +15,22 @@ use crate::core::rest::{EndpointSet, Unchecked};
...
+    interface_types: HashSet<String>,
+}
+
+impl ConfigModule {
+    pub fn config_mut(&mut self) -> &mut Config {
+        &mut self.config
+    }
+    pub fn set_extensions(&mut self, extensions: Extensions) {
+        self.extensions = extensions;
+    }
```
Comment (tusharmath) [inline src/core/config/config_module.rs:33]:
```suggestion
    pub fn set_config(self, config: Config) -> Config {
        let c = Self::from(config);
         c.extensions = self.extensions;
    }
    
    pub fn with_extensions(&mut self, extensions: Extensions) {
        self.extensions = extensions;
    }
```

Comment (be-hunter) [reply src/core/config/config_module.rs]: updated.

File: src/core/config/reader.rs [hunk 2]
```diff
@@ -131,16 +129,16 @@ impl ConfigReader {
...
                 }
             }
         }
 
         // Recreating the ConfigModule in order to recompute the values of
         // `input_types`, `output_types` and `interface_types`
-        let mut final_config_module = ConfigModule::from(config_module.config);
-        final_config_module.extensions = config_module.extensions;
+        let mut final_config_module = ConfigModule::from(config_module.config().clone());
+        final_config_module.with_extensions(extensions);
```
Comment (tusharmath) [inline src/core/config/reader.rs:141]: This is just a bit awkward. I think the main problem is that we are modifying the `config` above when there is a `LinkType::Config` and then we use `merge_right` on `ConfigModule`. 

I think a better approach would be to pull out the config from config_module modify it wherever required and then recreate the ConfigModule. We should not use the `merge_right` feature in ConfigModule, it's derived and is logically incorrect.

File: src/core/config/reader.rs [hunk 3]
```diff
@@ -131,16 +129,16 @@ impl ConfigReader {
...
         Ok(final_config_module)
```
Comment (tusharmath) [inline src/core/config/reader.rs:142]:
```suggestion
        Ok(ConfigModule::from(config).with_extensions(extensions))
```

This is a much better usage. Notice that the `config` passed to the `from` method. It is a config that is extracted in begining of the function and then modified whenever we hit a `LinkType::Config`.

Feel free to connect on discord if it's not clear.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**